videoretrieval/metrics/loss.py

```diff
@@ -0,0 +1,58 @@
+"""Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Defines the loss function needed to train the model.
+"""
+
+import tensorflow as tf
+
+def bidirectional_max_margin_ranking_loss(video_embeddings, text_embeddings, m):
+    """Implementation of the Bidirectional max margin ranking loss.
+
+    Arguments:
+        video_embeddings: a tensor of dimension (batch size x embedding space
+            size) where the ith element is the embedding for the ith video in
+            the batch
+        text_embeddings: a tensor of dimension (batch size x embedding space
+            size) where the ith element is the embedding for the ith caption in
+            the batch
+        m: a hyper parameter
```

Comment:
From style guide: "avoid single character names, except for specifically allowed cases." What hyper parameter is this. Ideally name should explain it. The comment is also not very helpful here.

Reply:
Totally. Just updated!